### PR TITLE
ENH: Added Copy metadata button to DICOM metadata widget

### PR DESCRIPTION
**Pull request by lassoan**
Added "Copy metadata" button (next to "Copy path" button) in the DICOM metadata browser (ctkDICOMObjectListWidget) to copy all metadata to the clipboard.
It is useful for collecting information from users when they reporting DICOM loading errors.

Also added tooltip for the existing, very useful, but hard-to-discover feature: if user clicks on a tag then its definition is displayed at http://dicomlookup.com.

### CODE REVIEW

**Comment by lassoan** [comment]
Example:

image

File: Libs/DICOM/Widgets/ctkDICOMObjectListWidget.cpp [hunk 1]
```diff
@@ -76,6 +77,44 @@ void ctkDICOMObjectListWidgetPrivate::setPathLabel(const QString& currentFile)
...
 
+// --------------------------------------------------------------------------
+QString ctkDICOMObjectListWidgetPrivate::dicomObjectModelAsString(QModelIndex parent /*=QModelIndex()*/, int indent /*=0*/)
+{
+  QString dump;
+  QString indentString;
+  for (int i = 0; i < indent; ++i)
+    {
+    indentString += "\t";
+    }
```
Comment (jcfr) [inline Libs/DICOM/Widgets/ctkDICOMObjectListWidget.cpp]:
```
QString("\t").repeated(indent);
```

**Comment by jcfr** [comment]
Should the data be copied as CSV to allow copy in spreadsheet ?

**Comment by lassoan** [comment]
Excel recognizes tab as a separator character, so you can copy-paste it.
However, the DICOM tags are unfortunately recognized as negative numbers by default (due to that they are numbers in parentheses). So, browsing in Excel is not working very well. I could change it to square brackets if you think that Excel browsing would be important.

File: Libs/DICOM/Widgets/ctkDICOMObjectListWidget.cpp (hunk 1 above)
Comment (lassoan) [inline Libs/DICOM/Widgets/ctkDICOMObjectListWidget.cpp]: Nice, thanks, I was looking for something like that but did not find it.

**Comment by jcfr** [comment]
> Excel recognizes tab as a separator character

I missed that, using tabs makes sense. Let's make sure we remove them from the values then.

**Comment by lassoan** [comment]
Data now shows up nicely in Excel:

image

**Comment by pieper** [comment]
👍

**Comment by lassoan** [comment]
@jcfr Can I update Slicer to use latest CTK? It seems there have been a couple of changes/improvements since the last version (console enhancements, etc)?

**Comment by jcfr** [comment]
I updated yesterday. You can definitively go ahead and update again.
Jc

**Pull request merged**